lib/chef/resource/locale.rb

```diff
@@ -67,10 +67,18 @@ def lc_all(arg = nil)
 
       action :update do
         description "Update the system's locale."
-        unless up_to_date?
-          converge_by "Updating System Locale" do
-            generate_locales unless unavailable_locales.empty?
-            update_locale
+        if windows?
+          unless new_resource.lang == get_system_locale
+            converge_if_changed :lang do
+              set_system_locale
+            end
+          end
+        else
+          unless up_to_date?
```

Comment:
there shouldn't be a `if windows?` check at this higher level.  the `up_to_date?` method should really work on both windows and unix, and the `if windows`? check should be pushed into that method.

similarly there should probably be a single `update_locale` or `set_system_locale` or whatever that worked on both windows and unix (and the generate_locales/update_locale complexity should be pushed down into the details of that method).